java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/RelationalSymbolicValue.java

```diff
@@ -188,18 +216,64 @@ public BooleanConstraint shouldNotInverse() {
     return states;
   }
 
-  @CheckForNull
-  private ProgramState checkRelation(BooleanConstraint booleanConstraint, ProgramState programState) {
+  private boolean checkRelation(BooleanConstraint booleanConstraint, ProgramState programState) {
     RelationState relationState = binaryRelation().resolveState(programState.getKnownRelations());
-    if (relationState.rejects(booleanConstraint)) {
-      return null;
+    return !relationState.rejects(booleanConstraint);
+  }
+
+  private List<SymbolicValue> transitiveRelations(ProgramState programState) {
+    BinaryRelation relation = binaryRelation();
+    return programState.getKnownRelations().stream()
+      .map(r -> r.deduceTransitiveOrSimplified(relation))
+      .filter(Objects::nonNull)
+      .map(RelationalSymbolicValue::binaryRelationToSymbolicValue)
+      .collect(Collectors.toList());
+  }
+
+  private static SymbolicValue binaryRelationToSymbolicValue(BinaryRelation binaryRelation) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 6, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1142)